### PR TITLE
Add kind demo deployment

### PR DESCRIPTION
**Pull request by nickolaev**
Signed-off-by: Nikolay Nikolaev <nikolay.nikolaev@konghq.com>

## PR Details

Add kind deployment folder for easier demonstration of Kuma.

### Description

A script to implement a step by step deployment of Kuma and the Demo.

### Motivation and Context

Provide a lightweight deployment option for exploring Kuma.

### Full changelog

* Adds the kind folder with the script and the kind cluster configuration

### CODE REVIEW

File: kind/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Kind Deployment Guide
+
+This is a simplified deployment of Kuma and the Demo application on top of [kind](https://kind.sigs.k8s.io/). It follows the steps described in [kubernetes](../kubernetes/README.md) folder, replacing the Minikube deployment.
+
+## Prerequisites
+
+There are only two prerequisites required before you run the demo: [docker](https://docs.docker.com/get-docker/) and [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
```
Comment (devadvocado) [inline kind/README.md:7]: if im not mistaken, user also needs to have kind installed?

Comment (nickolaev) [reply kind/README.md:7]: Actually no. The script will download the fixed version of KIND that is specified in it. I have just updated it to the lates 0.8.0 released a couple days ago.

File: kind/kind.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
+#/bin/sh
+
+KUMACTL_BIN=${HOME}/bin/kumactl
+KUMA_VERSION="0.4.0"
+
+KIND_BIN=${HOME}/bin/kind
+KIND_VERSION="v0.8.0"
```
Comment (devadvocado) [inline kind/kind.sh:7]: need to add a check of the version and force an upgrde

File: kind/kind.sh [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+# Create the kind cluster to host Kuma and the Demo
+${KIND_BIN} create cluster --name="${KIND_CLUSTER}" --config ./cluster.yaml --wait 120s
+until kubectl taint node ${KIND_CLUSTER}-control-plane node-role.kubernetes.io/master:NoSchedule- ;
+do
+  echo "Waiting for the cluster to come up" && sleep 3;
+done
+
+# Deploy Kuma and wait for the deployment to finish
+${KUMACTL_BIN} install control-plane | kubectl apply --wait -f -
+kubectl wait -n kuma-system --timeout=300s --for condition=Ready --all pods
```
Comment (devadvocado) [inline kind/kind.sh:91]: I get 
```
Error: unknown flag: --all
Examples:
  # Wait for the pod "busybox1" to contain the status condition of type "Ready".
  kubectl wait --for=condition=Ready pod/busybox1
...
```

File: kind/kind.sh (hunk 2 above)
Comment (nickolaev) [reply kind/kind.sh]: Done also for kumactl.

File: kind/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+default   kuma-demo-backend-v0-6fdb79ddfd-pg82b.kuma-demo   app=kuma-demo-backend env=prod pod-template-hash=6fdb79ddfd protocol=http service=backend.kuma-demo.svc:3001 version=v0     Online   4s                   3s                 4               0
+default   postgres-master-78d9c9c8c9-p7swm.kuma-demo        app=postgres pod-template-hash=78d9c9c8c9 protocol=tcp service=postgres.kuma-demo.svc:5432                                  Online   4s                   2s                 4               0
+default   redis-master-657c58c859-9r57p.kuma-demo           app=redis pod-template-hash=657c58c859 protocol=tcp role=master service=redis.kuma-demo.svc:6379 tier=backend               Online   6s                   4s                 4               0
+Kuma GUI is available at http://localhost:5683/
+Kuma DEMO is available at http://localhost:8080/
+Type 'quit' to exit.
+```
+
+As the message states, the Demo site can be found at [http://localhost:8080](http://locahost:8080) and the Kuma GUI at [http://localhost:5683](http://locahost:5683). You can also use `kubectl` and `${HOME}/bin/kumactl` to explore the deployment and configuration.
```
Comment (devadvocado) [inline kind/README.md:99]: should we point them to the demo instructions on how to apply policies found in the Kubernetes readme?

**Comment by devadvocado** [comment]
please update the master README to include kind

File: kind/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+This is a simplified deployment of Kuma and the Demo application on top of [kind](https://kind.sigs.k8s.io/). It follows the steps described in [kubernetes](../kubernetes/README.md) folder, replacing the Minikube deployment.
+
+## Prerequisites
+
+There are only two prerequisites required before you run the demo: [docker](https://docs.docker.com/get-docker/) and [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+## Running the demo
+
+Running the demo is easy, just type `./kind.sh` and the script will download, deploy and configure everything as described in [kubernetes](../kubernetes/README.md). In the end it will
```
Comment (devadvocado) [inline kind/README.md:11]: `In the end it will` seems to end very abruptly

Comment (nickolaev) [reply kind/README.md]: Fixed, thanks!

**Pull request merged**